### ISSUE TITLE
Can't annotate velocity vectors with SPH data in yt4.0

### ISSUE THREAD

**Issue by ibutsky** [yt-project/yt#2256]
### Bug report

**Bug summary**

When I make a projection or slice plot of Tipsy data with the yt-4.0 branch, I can't annotate velocity. I've tried both "annotate_quiver", "annotate_velocity", and "annotate_streamlines", with various aliases for the velocity fields. 

**Code for reproduction**

Dataset: "TipsyGalaxy" from the yt data hub. Download link:http://yt-project.org/data/TipsyGalaxy.tar.gz

```python
import yt
ds = yt.load('galaxy.00300')
v, cen = ds.h.find_max(('gas', 'density'))
p = yt.ProjectionPlot(ds, 'x', ('gas', 'temperature'),\
 center = cen, width = (200, 'kpc'), weight_field = ('gas', 'density'))
p.annotate_velocity()
p.save()

```

**Actual outcome**

```
# If applicable, paste the console output here
yt : [INFO     ] 2019-04-30 15:54:22,479 Parameters: current_time              = 20.100000000000044
yt : [INFO     ] 2019-04-30 15:54:22,479 Parameters: domain_dimensions         = [1 1 1]
yt : [INFO     ] 2019-04-30 15:54:22,479 Parameters: domain_left_edge          = None
yt : [INFO     ] 2019-04-30 15:54:22,479 Parameters: domain_right_edge         = None
yt : [INFO     ] 2019-04-30 15:54:22,479 Parameters: cosmological_simulation   = 0.0
yt : [INFO     ] 2019-04-30 15:54:22,513 Loading KDTree from galaxy.00300.kdtree
yt : [INFO     ] 2019-04-30 15:54:22,536 Allocating for 3.154e+05 particles
yt : [INFO     ] 2019-04-30 15:54:24,262 Max Value is 1.12117e-20 at 0.1224512308835983 -0.0390233956277370 0.3707219660282135
yt : [INFO     ] 2019-04-30 15:54:24,272 xlim = -100.039023 99.960977
yt : [INFO     ] 2019-04-30 15:54:24,272 ylim = -99.629278 100.370722
yt : [INFO     ] 2019-04-30 15:54:24,273 xlim = -100.039023 99.960977
yt : [INFO     ] 2019-04-30 15:54:24,273 ylim = -99.629278 100.370722
yt : [INFO     ] 2019-04-30 15:54:24,282 Making a fixed resolution buffer of (('gas', 'temperature')) 800 by 800
yt : [INFO     ] 2019-04-30 15:54:24,954 Making a fixed resolution buffer of (('gas', 'density')) 800 by 800
Traceback (most recent call last):
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_window.py", line 1126, in run_callbacks
    callback(cbw)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 71, in _check_geometry
    return func(self, plot)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 354, in __call__
    return qcb(plot)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 71, in _check_geometry
    return func(self, plot)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 446, in __call__
    fv_x = plot.data[self.field_x]
  File "/nobackupp2/ibutsky/software/yt/yt/data_objects/data_containers.py", line 268, in __getitem__
    rv = self.ds.arr(self.field_data[key], fi.units)
KeyError: 'velocity_y'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "plot.py", line 8, in <module>
    p.save()
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_container.py", line 92, in newfunc
    args[0]._setup_plots()
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_window.py", line 1075, in _setup_plots
    self.run_callbacks()
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_window.py", line 1132, in run_callbacks
    sys.exc_info()[2])
  File "/home4/scicon/tools/opt/sles12/python/2018.3.222/intelpython3/lib/python3.6/site-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_window.py", line 1126, in run_callbacks
    callback(cbw)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 71, in _check_geometry
    return func(self, plot)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 354, in __call__
    return qcb(plot)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 71, in _check_geometry
    return func(self, plot)
  File "/nobackupp2/ibutsky/software/yt/yt/visualization/plot_modifications.py", line 446, in __call__
    fv_x = plot.data[self.field_x]
  File "/nobackupp2/ibutsky/software/yt/yt/data_objects/data_containers.py", line 268, in __getitem__
    rv = self.ds.arr(self.field_data[key], fi.units)
yt.utilities.exceptions.YTPlotCallbackError: annotate_velocity callback failed with the following error: 'velocity_y'

```

**Expected outcome**

**Version Information**

  * Operating System: Pleiades, 4.4.162-94.72.1.20181113-nasa
  * Python Version: Python 3.6.8
  * yt version: yt-4.0 branch, commit c24cec7a6531b7960cb54b8a6d0a407537bbb4f3
  * Other Libraries (if applicable): 

Following the installation instructions for the demeshened version of yt/Trident here: https://nbviewer.jupyter.org/url/trident-project.org/notebooks/trident_demesh_install.ipynb

**Comment by MatthewTurk** [comment]
Hi @ibutsky !  Thanks for reporting this.  I *think* that the issue is that `velocity_x` is what we would call the grid or gridded fields, but we don't need that here.  I believe you may be able to utilize `annotate_vector_field` instead, and use `particle_velocity_x` and `particle_velocity_y` (or whatever the appropriate vector fields are.)  We may be able to fix this, but it might require some thought from @qobilidop or @jisuoqing , who are experts (respectively) on demeshening and plot annotations.

**Comment by Xarthisius** [comment]
As @MatthewTurk said above: `annotate_velocity` is shortcut for a class called `VelocityCallback` which is defined here: 
https://github.com/yt-project/yt/blob/14f330d5200c9e1e9036030433b3b677aef5c97c/yt/visualization/plot_modifications.py#L307

unfortunately this callback makes some assumptions about fields it uses for plotting, which can be seen here:

https://github.com/yt-project/yt/blob/master/yt/visualization/plot_modifications.py#L355-L370

**Comment by jisuoqing** [comment]
Previously I also thought it's an issue related to field naming, but now I realized that this error is caused by `pixelize_cartesian` routine which obviously does not work for particle data. The solution is to simply replace the pixelize routine with `ds.coordinates.pixelize`, as I did in PR #2105 for the `master` branch.

But since the `master` branch will get finally merged into the `yt-4.0`, do I need to perform the same fix in `master`, or just wait for branch merging?

@ibutsky For now, I think you could use the routine `annotate_line_integral_convolution`, which utilize the `ds.coordinates.pixelize` routine and therefore should work well with particle data.

**Comment by ibutsky** [comment]
Thanks, everyone! 

I tried using annotate_line_integral_convolution, and that did work.

**Comment by jisuoqing** [comment]
Thank you for reporting this Iryna! Other annotations (quiver, streamline, etc.) will definitely get fixed, at the latest when the `yt-4.0` gets merged into the `master` branch.

**Comment by chrishavlin** [comment]
just confirmed that `annotate_velocity` does indeed now work with tipsy, going to close. For posterity, the original code sample:

```python
import yt
ds = yt.load('TipsyGalaxy/galaxy.00300')
v, cen = ds.find_max(('gas', 'density'))
p = yt.ProjectionPlot(ds, 'x', ('gas', 'temperature'),\
 center = cen, width = (200, 'kpc'), weight_field = ('gas', 'density'))
p.annotate_velocity()
p.save()
```

now runs successfully:


![galaxy 00300_Projection_x_temperature_density](https://user-images.githubusercontent.com/22038879/144653883-b28cec95-e2c3-4714-bca9-20940fa6d44c.png)
